src/main/java/net/revelc/code/formatter/ValidateMojo.java

```diff
@@ -56,12 +74,27 @@ protected void doFormatFile(final File file, final ResultCollector rc, final Pro
         super.doFormatFile(file, rc, hashCache, basedirPath, true);
 
         if (rc.successCount != 0) {
-            throw new MojoFailureException("File '" + file
-                    + "' has not been previously formatted.  Please format file and commit before running validation!");
+            String errorMessage = String.format(
+                    "File '%s' has not been previously formatted. Please format file (for example by invoking `%s`) and commit before running validation!",
+                    file, formatCommand());
+            throw new MojoFailureException(errorMessage);
         }
         if (rc.failCount != 0) {
             throw new MojoExecutionException("Error formating '" + file + "' ");
         }
     }
 
+    private String formatCommand() {
+        String validateInvocation = String.format("mvn %s:%s:%s:%s", mojoGroupId, mojoArtifactId, mojoVersion,
+                FORMAT_MOJO_NAME);
+        boolean isMultiModule = mavenSession.getAllProjects().size() > 1;
+        if (isMultiModule) {
+            Artifact currentProjectArtifact = mavenSession.getCurrentProject().getArtifact();
+            return String.format("%s -pl '%s:%s'", validateInvocation, currentProjectArtifact.getGroupId(),
```

Comment:
`-f` would be quicker, especially for large projects like Quarkus. `MavenProject.getFile()` (or `.getBasedir()`) should give you what you'd need for that.